e2e/provision/gce_init.sh

```diff
@@ -36,13 +36,13 @@ apt-get install -y git
 cd /home/ubuntu
 
 runuser -u ubuntu git clone "$REPO" test-infra
-cd test-infra && runuser -u ubuntu git checkout "$BRANCH" && cd ..
+cd test-infra && runuser -u ubuntu -- git checkout -b "$BRANCH" --track "origin/$BRANCH" && cd ..
```

Comment:
Using this approach will result in running the latest merged code in main, versus running the current changes submitted into the PR.

Reply:
You can pass in the repo and branch, but that is for the *test-infra* repo and branch.

Because all our automations pull based on tag version numbers, I don't see how we get around this. I think we need to address that as a larger point, and it may be repo-by-repo. For example, if we want to run e2e on changes to the Nephio controller code or KRM function code that is in the nephio repo, we are doing to need to figure out how to push the image tags through to the packages that are installed via this mechanism. I don't think we address that properly right now in this version or prior versions of this code.